src/pyggdrasil/tree_inference/_mcmc.py

```diff
@@ -23,22 +26,27 @@ class Tree:
     Attrs:
         tree_topology: the topology of the tree
           encoded in the adjacency matrix.
+          No self-loops, i.e. diagonal is all zeros.
           Shape ``(N+1, N+1)``
         labels: maps nodes in the tree topology
           to the actual mutations.
           Note: the last position always maps to itself,
-          as it's the root and we use the convention
+          as it's the root, and we use the convention
           that root has the largest index.
           Shape ``(N+1,)``
-
-    TODO:
-        Decide whether the ``tree_topology`` should have ones
-          on the diagonal and adjust the docstring.
     """
 
     tree_topology: jax.Array
     labels: jax.Array
 
+    def to_TreeNode(self) -> TreeNode:
+        """Converts this Tree to a TreeNode.
+        Returns the root node of the tree."""
+        root = tree_inf.adjacency_to_root_dfs(self.tree_topology)
+        return root
+
+    # TODO: Pawel / Gordon: discuss Tree vs TreeNode - merge / interface / etc.
```

Comment:
The idea here is that `TreeNode` is in the public API and it could be used with multiple subpackages.

In this subpackage we have `Tree`, which is *an internal (private) representation* of a `TreeNode` in the format which makes it easy to do MCMC: namely, adjacency matrix encoding topology and the labels, determining the mapping from the tree topology to the mutations. It should be a part of the private API (i.e., we don't export it to any "visible" `__init__.py`) and, yes, conversion between these should be possible (but only for our internal purposes).

Reply:
OK, understood - makes sense. I will revert the `Tree` class back to private. A conversion is already implemented from `Tree` -> `TreeNode` mainly for easy visualization to check workings / debug.

Reply:
Wonderful! :slightly_smiling_face: